data/profile-physlr.Rmd

```diff
@@ -0,0 +1,384 @@
+---
+title: "Physlr profiling - time and memory"
+author: "Amirhossein Afshinfard"
+date: "2/11/2020"
+output: html_document
+params:
+  path:
+    label: "Path where the *.time files are stored"
+    value: "./"
+    input: text
+---
+
+
+```{r setup, include=FALSE}
+#Rscript -e 'rmarkdown::render("$(physlr_path)/data/profile-code.rmd", "html_document", "$*.prof.html", knit_root_dir="$(PWD)", #output_dir="$(PWD)", params=list(path="$(physlr_path)/data/"))'
+knitr::opts_chunk$set(echo = TRUE)
+list.of.packages <- c("dplyr", "ggplot2", "stringr")
+new.packages <- list.of.packages[!(list.of.packages %in% installed.packages()[,"Package"])]
+if(length(new.packages)) install.packages(new.packages)
+
+library("dplyr")
+library("ggplot2")
+library("stringr")
+
+knit_print.data.frame <- function(x, ...) kable(x) %>% paste(collapse = "\n") %>% asis_output
+path <- params$path
+
+```
+
+## General information:
+
+Profiling of Physlr steps, based on reports provided by time command stored in *.time files in the input directory.
+
+```{r echo=FALSE, message=FALSE, warning=FALSE}
+
+m <- matrix(ncol = 7)
+df = data.frame(m)
+colnames(df)=c("step",
+               "total_time_min",
+               "time_hr",
+               "time_min",
+               "time_sec",
+               "mem",
+               "pl")
+
+#path <- "/projects/btl/jowong/github/physlr/data_hg004_stlfr/v2release"
+
+if (substr(path,nchar(path),nchar(path)) != "/")
+  path <- paste(path,"/",sep="")
+ldir <-  normalizePath(path)
+
+
+```
+
+The profiling directory including *.time files:
+
+```{r echo=FALSE}
+cat(ldir)
+
+finf <- file.info(dir(path = ldir, full.names = TRUE), extra_cols = FALSE)
+sorted_finf <- finf[with(finf, order(as.POSIXct(ctime))), ]
+sorted_files <- rownames(sorted_finf)
+sorted_time_files <- sorted_files[substr(sorted_files,
+                                         nchar(sorted_files)-4,nchar(sorted_files)) == ".time"]
+
+# Or alternatively:
+#file.info("/projects/btl/jowong/github/physlr/data_hg004_stlfr/v2release/hg004_k40.histogram.time")
```

Comment:
Was this an alternate approach you tried and decided against? We just don't want to have commented-out code